File: WondayWall.Android/app/src/main/kotlin/com/studiofreesia/wondaywall/models/DebugConfig.kt
package com.studiofreesia.wondaywall.models

import kotlinx.serialization.Serializable

// デバッグ専用設定モデル
@Serializable
data class DebugConfig(
    // ダミーAIサービスに切り替える（切り替えは次回起動時に反映）
    val useDummyAiService: Boolean = false,
    // ダミー実装でのプロンプト生成遅延秒数
    val dummyPromptDelaySeconds: Int = 180,
    // ダミー実装での画像生成遅延秒数
    val dummyImageDelaySeconds: Int = 600,
    // ダミー実装で生成コンテキストに含めるニュース件数
    val dummyNewsCount: Int = 4,
) {
    fun normalized(): DebugConfig = copy(
        dummyPromptDelaySeconds = dummyPromptDelaySeconds.coerceIn(MIN_DELAY_SECONDS, MAX_DELAY_SECONDS),
        dummyImageDelaySeconds = dummyImageDelaySeconds.coerceIn(MIN_DELAY_SECONDS, MAX_DELAY_SECONDS),
        dummyNewsCount = dummyNewsCount.coerceIn(MIN_NEWS_COUNT, MAX_NEWS_COUNT),
    )

    companion object {
        const val MIN_DELAY_SECONDS = 1
        const val MAX_DELAY_SECONDS = 3600
        const val MIN_NEWS_COUNT = 0
        const val MAX_NEWS_COUNT = 20
    }
}

File: WondayWall/Models/DebugConfig.cs
namespace WondayWall.Models;

#if DEBUG
/// <summary>繝・ヰ繝・げ繝薙Ν繝牙ｰら畑縺ｮ讀懆ｨｼ逕ｨ險ｭ螳壹ゅΜ繝ｪ繝ｼ繧ｹ繝薙Ν繝峨↓縺ｯ蜷ｫ繧√↑縺・・/summary>
public class DebugConfig
{
    private const int MinDelaySeconds = 1;
    private const int MaxDelaySeconds = 3600;
    private const int MinNewsCount = 0;
    private const int MaxNewsCount = 20;

    public bool UseDummyAiService { get; set; } = false;
    public int DummyPromptDelaySeconds { get; set; } = 180;
    public int DummyImageDelaySeconds { get; set; } = 600;
    public int DummyNewsCount { get; set; } = 4;

    public void Normalize()
    {
        DummyPromptDelaySeconds = Math.Clamp(DummyPromptDelaySeconds, MinDelaySeconds, MaxDelaySeconds);
        DummyImageDelaySeconds = Math.Clamp(DummyImageDelaySeconds, MinDelaySeconds, MaxDelaySeconds);
        DummyNewsCount = Math.Clamp(DummyNewsCount, MinNewsCount, MaxNewsCount);
    }
}
#endif
